### PR TITLE
[6011] Content can't be unlocked or edited after Save & Close and Save & Minimize action

### PR DESCRIPTION
**Pull request by jvega190**
- Disable submit button when already submitting.
- Disable closeDialog when submitting.
- Replace confirm dialog with the React one.

https://github.com/craftercms/craftercms/issues/6011

### CODE REVIEW

File: ui/app/src/pages/LegacyMultiChoiceSaveButton.tsx
Comment (rart) [inline ui/app/src/pages/LegacyMultiChoiceSaveButton.tsx:1]: We discussed not to putting this on the react source, only if **strictly** necessary. What happened?

Comment (jvega190) [reply ui/app/src/pages/LegacyMultiChoiceSaveButton.tsx:1]: Moved to forms-engine

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 1]
```diff
@@ -1927,6 +1931,55 @@ var CStudioForms =
...
+                //       label: formatMessage(formEngineMessages.saveAndClose),
+                //       callback: (e) => onMultiChoiceSaveButtonClick(e, 'saveAndClose')
+                //     },
+                //     {
+                //       id: 'saveAndMinimize',
+                //       label: formatMessage(formEngineMessages.saveAndMinimize),
+                //       callback: (e) => onMultiChoiceSaveButtonClick(e, 'saveAndMinimize')
+                //     }
+                //   ]
+                // });
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js]: ???

File: ui/app/src/components/LegacyFormDialog/LegacyFormDialog.tsx [hunk 2]
```diff
@@ -48,6 +53,27 @@ export function LegacyFormDialog(props: LegacyFormDialogProps) {
...
+            dispatch(updateEditConfig({ disableOnClose: false }));
+            break;
+          }
+        }
+      });
+
+    return () => {
+      messagesSubscription.unsubscribe();
+    };
+  }, [dispatch]);
```
Comment (rart) [inline ui/app/src/components/LegacyFormDialog/LegacyFormDialog.tsx]: This should go in EmbeddedLegacyContainer where the rest of events are handled

File: ui/app/src/state/reducers/dialogs/edit.ts [hunk 3]
```diff
@@ -27,7 +27,8 @@ const initialState: LegacyFormDialogStateProps = {
   path: null,
   site: null,
   authoringBase: null,
-  inProgress: true
+  inProgress: true,
+  disableOnClose: false
```
Comment (rart) [inline ui/app/src/state/reducers/dialogs/edit.ts]: Let's rename to isSubmitting, for consistency.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**